catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java

```diff
@@ -113,6 +119,14 @@ protected void onCreate(Bundle savedInstanceState) {
 			return;
 		}
 
+		if (getIntent().getBooleanExtra(RESTART_INTENT, false)) {
+			Intent restartIntent = getApplicationContext().getPackageManager().getLaunchIntentForPackage(getApplicationContext().getPackageName());
+			PendingIntent intent = PendingIntent.getActivity(getApplicationContext(), 0, restartIntent, PendingIntent.FLAG_CANCEL_CURRENT);
+			AlarmManager manager = (AlarmManager) getApplicationContext().getSystemService(Context.ALARM_SERVICE);
+			manager.set(AlarmManager.RTC, System.currentTimeMillis() + restartDelayAfterAppClose, intent);
```

Comment:
maybe have a look at recreate(): https://developer.android.com/reference/android/app/Activity.html#recreate%28%29

Disclaimer: I haven't spent much time digging into this, but it may be more appropriate here